### PR TITLE
Feature/spark line

### PR DESCRIPTION
**Pull request by jryu01**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -107,6 +107,19 @@ export function findColumn(dataView: DataView, dataRoleName: string, multi?: boo
 }
 
 /**
+ * Check if provided dataView has all the columns with given data role names.
+ *
+ * @export
+ * @param   {DataView} dataView      A Powerbi dataView object.
```
Comment (psmacmur) [inline src/utils.ts]: comment nitpick: should be "PowerBI"

File: capabilities.json [hunk 2]
```diff
@@ -39,6 +39,12 @@
       "description": "Each facet instance is segmented by this field."
     },
     {
+      "name": "sparklineData",
+      "kind": 0,
+      "displayName": "Sparkline Axis",
+      "description": "A field to place on the horizontal axis for supporting spark line for the facet instance."
```
Comment (psmacmur) [inline capabilities.json]: comment nitpick: maybe "A field to render on the horizontal axis as a sparkline" ?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "essex.widget.facetkey",
-  "version": "0.3.11",
+  "version": "0.4.12",
```
Comment (psmacmur) [inline package.json]: are you sure we're bumping from 0.3.x to 0.4.x already?

File: src/utils.ts [hunk 4]
```diff
@@ -152,6 +165,27 @@ export function createSegments(bucket: any, mainColor: string, isHighlight: bool
     }));
 }
 
+
+
+/**
+ * Creates a timeseries data for facet sparkline.
```
Comment (psmacmur) [inline src/utils.ts]: comment nitpick: since "data" is plural, consider "Creates timeseries data for a facet sparkline"

File: src/utils.ts [hunk 5]
```diff
@@ -152,6 +165,27 @@ export function createSegments(bucket: any, mainColor: string, isHighlight: bool
...
+ *
+ * @export
+ * @param {any[]}           sparklineXDomain A domain of x values of the sparkline data.
+ * @param {Object}          sparklineData    Sparkline data that contains facet instacne and highlight count that mapped to x values.
```
Comment (psmacmur) [inline src/utils.ts]: typo & grammar: should probably be "Sparkline data that contains facet instance and highlight counts mapped to x values."

File: package.json (hunk 3 above)
Comment (jryu01) [inline package.json]: I think we can bump minor version since a new feature is added.

**Pull request merged**